mkdocs/overrides/devbook/transactions/bonded-aggregate.py

```diff
@@ -0,0 +1,231 @@
+	print('Creating hash lock transaction...')
+	hash_lock = facade.transaction_factory.create({
+		'type': 'hash_lock_transaction_v1',
+		'signer_public_key': account_a_key_pair.public_key,
+		'deadline': timestamp.add_hours(2).timestamp,
+		'mosaic': {
+			'mosaic_id': generate_mosaic_alias_id('symbol.xym'),
+			'amount': 10_000_000  # 10 XYM deposit
+		},
+		'duration': 100,  # Lock duration in blocks
+		'hash': bonded_hash
+	})
+	hash_lock.fee = Amount(fee_mult * hash_lock.size)
+
+	# Sign and announce hash lock
+	hash_lock_signature = facade.sign_transaction(
+		account_a_key_pair, hash_lock)
+	hash_lock_payload = facade.transaction_factory.attach_signature(
+		hash_lock, hash_lock_signature)
+	hash_lock_hash = facade.hash_transaction(hash_lock)
+	print(f'Hash lock transaction hash: {hash_lock_hash}')
+
+	announce_path = '/transactions'
+	print(f'Announcing hash lock to {announce_path}')
+	announce_request = urllib.request.Request(
+		f'{NODE_URL}{announce_path}',
+		data=hash_lock_payload.encode(),
+		headers={'Content-Type': 'application/json'},
+		method='PUT'
+	)
+	with urllib.request.urlopen(announce_request) as response:
+		print(f'  Response: {response.read().decode()}')
+
+	# Wait for hash lock confirmation
+	print(f'Waiting for hash lock confirmation...')
+	status_path = f'/transactionStatus/{hash_lock_hash}'
+	for attempt in range(60):
+		time.sleep(1)
+		try:
+			with urllib.request.urlopen(
+				f'{NODE_URL}{status_path}'
+			) as response:
+				status = json.loads(response.read().decode())
+				if status['group'] == 'confirmed':
+					print(f'  Hash lock confirmed')
+					break
+		except urllib.error.HTTPError as e:
+			print(f'  Hash lock status: unknown | Cause: ({e.msg})')
+
+	# Announce bonded transaction
+	partial_path = '/transactions/partial'
+	print(f'Announcing bonded transaction to {partial_path}')
+	partial_request = urllib.request.Request(
+		f'{NODE_URL}{partial_path}',
+		data=bonded_json_payload.encode(),
+		headers={'Content-Type': 'application/json'},
+		method='PUT'
+	)
+	with urllib.request.urlopen(partial_request) as response:
+		print(f'  Response: {response.read().decode()}')
+
+	# Wait for transaction to reach partial status
+	print('Waiting for bonded transaction to reach partial status...')
+	status_path = f'/transactionStatus/{bonded_hash}'
+	for attempt in range(60):
+		time.sleep(1)
+		try:
+			with urllib.request.urlopen(
+				f'{NODE_URL}{status_path}'
+			) as response:
+				status = json.loads(response.read().decode())
+				if status['group'] == 'partial':
+					print(
+						'  Transaction is partial, '
+						'ready for cosignatures'
+					)
+					break
+		except urllib.error.HTTPError as e:
+			print(f'  Transaction status: unknown | Cause: ({e.msg})')
+
+	# Submit Account B's cosignature
+	print('Submitting Account B\'s cosignature...')
+	cosignature_path = '/transactions/cosignature'
+	cosignature = facade.cosign_transaction(
+		account_b_key_pair, bonded_transaction, True)
```

Comment:
If you use `bonded_transaction` directly, it means that Account B has had access to it off-chain.
This is OK for the Complete Aggregate tutorial, but I think this one should do everything on-chain.

You knew I was going to ask for this :D

So, at the very least, the transaction to cosign should be retrieved from the chain, using its hash (which would be provided off-chain).
But even better, Account B can monitor partial transactions waiting for its signature using /transactions/partial (GET).
This is an example, so there is no need to poll (we know the partial transaction will be there at this point),
but the text should explain that Account B can periodically check if there are partial transactions waiting for its signature.